test/e2e/framework/services.go

```diff
@@ -47,3 +47,16 @@ func (f *Framework) CreateTCPService(cluster ClusterIndex, selectorName string,
 		return service, err
 	}, NoopCheckResult).(*v1.Service)
 }
+
+// AwaitServiceByAnnotation queries the service and looks for the presence of annotation.
+func (f *Framework) AwaitServiceByAnnotation(cluster ClusterIndex, annotation string, svcName string, namespace string) *v1.Service {
+	return AwaitUntil("get"+annotation+" annotation for service "+svcName, func() (interface{}, error) {
+		return f.ClusterClients[cluster].CoreV1().Services(namespace).Get(svcName, metav1.GetOptions{})
```

Comment:
It seems we check for `IsNotFound` and not assume the Service exists yet to be complete, ie:

```
service, err := f.ClusterClients[cluster].CoreV1().Services(namespace).Get(svcName, metav1.GetOptions{})
if apierrors.IsNotFound(err) {
    return nil, nil
}
return service, err

```

Reply:
Done, thanks for reviewing @tpantelis